django/river/common/analyzer/analyzer.py

```diff
@@ -39,23 +44,39 @@ def load_analysis(self, batch_id, resource_id) -> Optional[Analysis]:
         cache_key = f"{batch_id}:{resource_id}"
         return self.analyses.get(cache_key)
 
-    def analyze(self, resource_mapping):
+    def analyze(self, resource_id, mappings):
         self._cur_analysis = Analysis()
 
+        # Retrieve col data from mappings
+        self._owners_data, self._columns_data = self.retrieve_col_data(mappings)
         # Analyze the mapping
-        self.analyze_mapping(resource_mapping)
+        self.analyze_mapping(resource_id, mappings)
 
         return self._cur_analysis
 
-    def analyze_mapping(self, resource_mapping):
+    def retrieve_col_data(self, mappings):
+        # Flatten owners->columns to search through all columns
+        columns_by_id: Dict[str, Dict[str, Any]] = {}
+        owner_names_by_id: Dict[str, str] = {}
+        for owner in mappings["credential"]["owners"]:
+            owner_names_by_id[owner["id"]] = owner["name"]
+            for column in owner["columns"]:
+                columns_by_id[column["id"]] = {**column, "owner": owner["name"]}
+
+        return owner_names_by_id, columns_by_id
+
+    def analyze_mapping(self, resource_id, mappings):
+        resource_mapping = next(mapping for mapping in mappings["resources"] if mapping["id"] == resource_id)
+
         self._cur_analysis.primary_key_column = self.get_primary_key(resource_mapping)
-        self._cur_analysis.source_id = resource_mapping["source"]["id"]
-        self._cur_analysis.source_credentials = resource_mapping["source"]["credential"]
-        self._cur_analysis.resource_id = resource_mapping["id"]
-        self._cur_analysis.definition_id = resource_mapping.get("definitionId")
-        self._cur_analysis.definition = resource_mapping.get("definition")
+        self._cur_analysis.source_id = mappings["id"]
+        self._cur_analysis.source_credentials = {k: v for k, v in mappings["credential"].items() if k != "owners"}
+        self._cur_analysis.resource_id = resource_id
+        self._cur_analysis.definition_id = resource_mapping.get("definition_id")
+        # FIXME: definition field is absent from mapping
+        # self._cur_analysis.definition = resource_mapping.get("definition")
```

Comment:
this deserves an issue, can you make a quick one please ? add a link to this line
EDIT: done https://github.com/arkhn/fhir-river/issues/595.